### DISCUSSION TITLE
Not showing layer number during print any more. How to fix?

### DISCUSSION THREAD

**Discussion by rschaeuble** [KlipperScreen/KlipperScreen#1523] [Q&A] [answered]
During print, KlipperScreen does not show the layer number any more (see screenshot).
MainSail still correctly shows it, so the info is inside the Gcode file.

Any ideas what could be wrong here?

![IMG_5979](https://github.com/user-attachments/assets/46ce5b7b-39c7-4aac-9b52-392c94de186e)

**Comment by alfrix** [comment, marked as answer]
klipperscreen will not calculate the layer based on height, as it can be very inaccurate

without the gcode and logs i cannot confirm that it has needed macro calls to show the current layer,

you need ensure the slicer is configured to provide the correct info to the firmware

https://klipperscreen.github.io/KlipperScreen/Quicktips/#layer-progress

**Reply by rschaeuble** [reply]
That was probably the problem: no SET_PRINT_STATS_INFO in the Gcode. I had this in the past; no idea where it went.
And as MainSail shows the current layer, I thought the Gcode was good.

Thanks!

**Reply by rschaeuble** [reply]
I verified it with a new print. It was indeed the missing SET_PRINT_STATS_INFO.
Thanks, @alfrix.
